src/App.tsx

```diff
@@ -1,61 +1,153 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { useState } from 'react';
+import usersFromServer from './api/users.json';
+import todosFromServer from './api/todos.json';
+import { Todo } from './types/todos';
+import { TodoList } from './components/TodoList';
+
+const getUser = (userId: number) => {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+};
+
+const initialTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+const getMaxId = (todos: Todo[]) => {
+  const maxId = Math.max(
+    ...todos.map(todo => todo.id),
+  );
+
+  return maxId + 1;
+};
 
 export const App = () => {
+  const [todos, setTodos] = useState(initialTodos);
+
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleErros] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [selectFieldError, setSelectFieldError] = useState(false);
+
+  const validRegex = /^[A-Za-z\u0400-\u04FF0-9\s]+$/;
+
+  const inputValidation = (text: string) => {
+    return text.trim().match(validRegex);
+  };
+
+  const handleTitleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newTitle = event.target.value;
+
+    setTitle(newTitle);
+
+    setTitleErros(!newTitle);
+  };
+
+  const handleSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setSelectFieldError(false);
+  };
+
+  const addTodo = (NewTodo: Todo) => {
+    setTodos(currentTodos => [...currentTodos, NewTodo]);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+    setTitleErros(false);
+    setSelectFieldError(false);
+  };
+
+  function handleSumbit(event: React.FormEvent) {
+    event.preventDefault();
+
+    setTitleErros(!inputValidation(title));
+    setSelectFieldError(!userId);
+
+    if (!inputValidation(title) || !userId) {
+      return;
+    }
+
+    addTodo({
+      id: getMaxId(todos),
+      title,
+      userId,
+      completed: false,
+      user: getUser(userId),
+    });
+
+    reset();
+  }
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form
+        action="/api/todos"
+        method="POST"
+        onSubmit={handleSumbit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="title-id">
+            Title:&nbsp;
+          </label>
+
+          <input
+            value={title}
+            id="title-id"
+            placeholder="Enter a title"
+            type="text"
+            data-cy="titleInput"
+            onChange={handleTitleInput}
+          />
+
+          {titleError && (
+            <span className="error">
+              {!title.trim()
```

Comment:
please don't call a function from the template because it will be executed every time when the state change, so make sense to move it to the variable